pinot-query-planner/src/main/java/org/apache/pinot/query/planner/physical/PinotDispatchPlanner.java

```diff
@@ -75,6 +76,19 @@ public DispatchableSubPlan createDispatchableSubPlan(SubPlan subPlan) {
     return finalizeDispatchableSubPlan(rootFragment, context);
   }
 
+  private Set<String> resolveActualTableNames(Set<String> tableNames) {
```

Comment:
Do we need to resolve table names here given the table name is already resolved in the scan node?

Reply:
I'm afraid it is, because under some circumstances `SubPlanMetadata` receives the table names directly from Calcite's `RelRoot` which still preserves the original table names provided by the user and not the actual, resolved ones.

Reply:
Consider adding some comments explaining this